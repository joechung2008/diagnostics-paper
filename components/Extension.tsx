import React from "react";
import { ScrollView, Text } from "react-native";
import Configuration from "../components/Configuration";
import StageDefinition from "../components/StageDefinition";
import { useExtensionStyles } from "../hooks/useExtensionStyles";
import type { ExtensionProps } from "../lib/types";

const Extension = ({
  config,
  extensionName,
  stageDefinition,
}: ExtensionProps) => {
  const dynamicStyles = useExtensionStyles();

  return (
    <ScrollView
      style={dynamicStyles.scrollView}
      contentContainerStyle={dynamicStyles.container}
    >
      <Text style={dynamicStyles.title}>{extensionName}</Text>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </ScrollView>
  );
};

export default Extension;
