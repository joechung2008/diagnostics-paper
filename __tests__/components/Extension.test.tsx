import { render } from "@testing-library/react-native";
import { MD3LightTheme, PaperProvider } from "react-native-paper";
import Extension from "../../components/Extension";

describe("Extension", () => {
  it("renders correctly with full data", () => {
    const mockExtension = {
      extensionName: "Test Extension",
      manageSdpEnabled: true,
      config: {
        setting1: "value1",
        setting2: "value2",
      },
      stageDefinition: {
        stage1: ["step1", "step2"],
        stage2: ["step3"],
      },
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Extension {...mockExtension} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with minimal data", () => {
    const mockExtension = {
      extensionName: "Minimal Extension",
      manageSdpEnabled: false,
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Extension {...mockExtension} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
