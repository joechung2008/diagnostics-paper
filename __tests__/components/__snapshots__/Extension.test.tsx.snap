// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extension renders correctly with full data 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#FFFBFE",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#1C1B1F",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Test Extension
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFBFE",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Configuration
      </Text>
      <View
        style={
          {
            "backgroundColor": "#FFFBFE",
            "borderColor": "#79747E",
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E7E0EC",
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1C1B1F",
                "flex": 1,
                "fontFamily": "System",
                "fontWeight": "bold",
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            Key
          </Text>
          <Text
            style={
              {
                "color": "#1C1B1F",
                "flex": 1,
                "fontFamily": "System",
                "fontWeight": "bold",
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            Value
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            setting1
          </Text>
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            value1
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            setting2
          </Text>
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            value2
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFBFE",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Stage Definitions
      </Text>
      <View
        style={
          {
            "backgroundColor": "#FFFBFE",
            "borderColor": "#79747E",
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E7E0EC",
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1C1B1F",
                "flex": 1,
                "fontFamily": "System",
                "fontWeight": "bold",
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            Key
          </Text>
          <Text
            style={
              {
                "color": "#1C1B1F",
                "flex": 1,
                "fontFamily": "System",
                "fontWeight": "bold",
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            Value
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            stage1
          </Text>
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            step1, step2
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#79747E",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "paddingBottom": 5,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            stage2
          </Text>
          <Text
            style={
              [
                {
                  "flex": 1,
                  "minWidth": 0,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                {
                  "color": "#1C1B1F",
                  "flex": 1,
                  "flexShrink": 1,
                  "fontFamily": "System",
                },
              ]
            }
          >
            step3
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Extension renders correctly with minimal data 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#FFFBFE",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#1C1B1F",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Minimal Extension
    </Text>
  </View>
</RCTScrollView>
`;
